src/main/java/org/hyperledger/fabric/sdk/Endpoint.java

```diff
@@ -385,90 +475,122 @@ private void addNettyBuilderProps(NettyChannelBuilder channelBuilder, Properties
                 }
             }
 
-            Utils.invokeMethod(channelBuilder, methodName, classParms, parmsArray);
+            try {
+                Utils.invokeMethod(channelBuilder, methodName, classParms, parmsArray);
+            } catch (IllegalAccessException | NoSuchMethodException | InvocationTargetException e) {
+                logger.error(format("Endpoint %s, exception '%s'", url, e.getMessage()), e);
+                logger.error(e);
+                throw new RuntimeException(e);
+            }
 
             if (logger.isTraceEnabled()) {
                 logger.trace(format("Endpoint with url: %s set managed channel builder method %s (%s) ", url,
-                        methodName, Arrays.toString(parmsArray)));
+                    methodName, Arrays.toString(parmsArray)
+                ));
 
             }
 
         }
 
     }
 
-    AbstractMap.SimpleImmutableEntry<PrivateKey, X509Certificate[]> getClientTLSProps(Properties properties) {
+    private Optional<Pair<PrivateKey, X509Certificate[]>> readClientTLSProps(String url, Properties properties) {
+        if (properties == null) {
+            return Optional.empty();
+        }
 
         // check for mutual TLS - both clientKey and clientCert must be present
-        byte[] ckb = null, ccb = null;
-        if (properties.containsKey(NetworkConfig.CLIENT_KEY_FILE) && properties.containsKey(NetworkConfig.CLIENT_KEY_BYTES)) {
-            throw new RuntimeException("Properties \"clientKeyFile\" and \"clientKeyBytes\" must cannot both be set");
-        } else if (properties.containsKey(NetworkConfig.CLIENT_CERT_FILE) && properties.containsKey(NetworkConfig.CLIENT_CERT_BYTES)) {
-            throw new RuntimeException("Properties \"clientCertFile\" and \"clientCertBytes\" must cannot both be set");
-        } else if (properties.containsKey(NetworkConfig.CLIENT_KEY_FILE) || properties.containsKey(NetworkConfig.CLIENT_CERT_FILE)) {
-            if ((properties.getProperty(NetworkConfig.CLIENT_KEY_FILE) != null) && (properties.getProperty(NetworkConfig.CLIENT_CERT_FILE) != null)) {
+        validatePropertyExclusivity(properties, CLIENT_KEY_FILE, CLIENT_KEY_BYTES);
+        validatePropertyExclusivity(properties, CLIENT_CERT_FILE, CLIENT_CERT_BYTES);
+
+        byte[] ckb;
+        byte[] ccb;
+        if (properties.containsKey(CLIENT_KEY_FILE) || properties.containsKey(CLIENT_CERT_FILE)) {
+            String propertyClientKeyFile = properties.getProperty(CLIENT_KEY_FILE);
+            String propertyClientCertFile = properties.getProperty(CLIENT_CERT_FILE);
+
+            if (propertyClientKeyFile != null && propertyClientCertFile != null) {
```

Comment:
You could reverse the logic here and move the throw from the else block this if block, similar to the short-circuit return you used at the top of this method. This would avoid the need for the following code to be in a nested block